simtools/data_model/validate_data.py

```diff
@@ -65,14 +70,68 @@ def validate_data_file(self):
 
         """
 
-        self._logger.info(f"Reading data from {self._data_file_name}")
-        self.data_table = Table.read(self._data_file_name, guess=True, delimiter=r"\s")
+        try:
+            if self._data_file_name.find("yaml") > 0 or self._data_file_name.find("yml") > 0:
+                self.data = gen.collect_data_from_yaml_or_dict(self._data_file_name, None)
+                self._logger.info("Reading data from yaml file: %s", self._data_file_name)
+            else:
+                self.data_table = Table.read(self._data_file_name, guess=True, delimiter=r"\s")
+                self._logger.info("Reading tabled data from file: %s", self._data_file_name)
+        except AttributeError:
+            pass
+
+    def _validate_data_dict(self):
+        """
+        Validate values. Creates first astropy table from data dict and then uses the same
+        methods as for tabled data.
+
+        """
+
+        try:
+            self._reference_data_columns = self._read_validation_schema(
+                self._schema_file_name, self.data["name"]
+            )
+            _quantities = []
+            for value, unit in zip(self.data["value"], self.data["units"]):
+                try:
+                    _quantities.append(value * u.Unit(unit))
+                except ValueError:
+                    _quantities.append(value)
+            self.data_table = Table(rows=[_quantities])
+            self.data_table.meta["name"] = self.data["name"]
+        except KeyError as exc:
+            raise KeyError("Data dict does not contain a 'name' or 'value' key.") from exc
+
+        if self._reference_data_columns is not None:
+            self._validate_data_columns()
+
+    def _validate_data_table(self):
+        """
+        Validate tabled data.
```

Comment:
"tabled data" reads a bit strange. Should it be "tabulated data" or just "table data"? Google doesn't give many results for "tabled data" so I think it's not just me.